dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -538,35 +698,180 @@ private static Deposit publishDepositOnZenodo(ActionsApi actionsApi, int deposit
+        AccessLinksApi accessLinksApi = new AccessLinksApi(createDockstoreZenodoClient());
+        try {
+            AccessLink accessLink = accessLinksApi.createAccessLink(recordId, new LinkPermissionSettings().permission(PermissionEnum.EDIT));
+            // Save the access link to the concept DOI because this link can edit ALL versions
+            dockstoreConceptDoi.setEditAccessLinkId(accessLink.getId());
+            workflow.getConceptDois().put(DoiCreator.DOCKSTORE, dockstoreConceptDoi);
+            return accessLink;
+        } catch (ApiException e) {
+            LOG.error("Could not create edit access link on Zenodo. Error is {}", e.getMessage(), e);
+            throw new CustomWebApplicationException("Could not create edit access link on Zenodo."
+                    + " Error is " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
+    }
 
-        public String getDoiAlias() {
-            return doiAlias;
+    /**
+     * Gets an existing access link.
+     * @param workflow
+     */
+    public static AccessLink getAccessLink(Workflow workflow) {
+        String existingDockstoreVersionDoiName = checkAndGetDockstoreVersionDoiName(workflow);
+        Doi dockstoreConceptDoi = workflow.getConceptDois().get(DoiCreator.DOCKSTORE);
+        checkAccessLinkExists(dockstoreConceptDoi);
+
+        String recordId = extractRecordIdFromDoi(existingDockstoreVersionDoiName);
+        AccessLinksApi accessLinksApi = new AccessLinksApi(createDockstoreZenodoClient());
+        try {
+            return accessLinksApi.getAccessLink(recordId, dockstoreConceptDoi.getEditAccessLinkId());
+        } catch (ApiException e) {
+            LOG.error("Could not get edit access link on Zenodo. Error is {}", e.getMessage(), e);
+            throw new CustomWebApplicationException("Could not get edit access link on Zenodo."
+                    + " Error is " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
+    }
 
-        public String getDoiUrl() {
-            return doiUrl;
+    /**
+     * Deletes the access link with edit permissions for the workflow.
+     * @param workflow
+     */
+    public static void deleteAccessLink(Workflow workflow) {
+        String existingDockstoreVersionDoiName = checkAndGetDockstoreVersionDoiName(workflow);
+        Doi dockstoreConceptDoi = workflow.getConceptDois().get(DoiCreator.DOCKSTORE);
+        checkAccessLinkExists(dockstoreConceptDoi);
+
+        String recordId = extractRecordIdFromDoi(existingDockstoreVersionDoiName);
+        AccessLinksApi accessLinksApi = new AccessLinksApi(createDockstoreZenodoClient());
+        try {
+            accessLinksApi.deleteAccessLink(recordId, dockstoreConceptDoi.getEditAccessLinkId());
+            // The access link is stored by the concept DOI. Set the values to null
+            dockstoreConceptDoi.setEditAccessLinkId(null);
+            workflow.getConceptDois().put(DoiCreator.DOCKSTORE, dockstoreConceptDoi);
```

Comment:
Is this needed? It doesn't look it would hurt, but wondering why it's here.

Reply:
You're right it's not needed, removed